pkgs/by-name/op/openstreetmap-carto/package.nix

```
@@ -0,0 +1,210 @@
+        inherit hash;
+      };
+    };
+  simplified_water_polygons = mkArchive {
+    url = "https://osmdata.openstreetmap.de/download/simplified-water-polygons-split-3857.zip";
+    archive_time = "20241201150706";
+    hash = "sha256-vVVbHBCbeaTaYAZRvIEQ5N1Dy5E+RaOSgZS3Uf59QkU=";
+  };
+  water_polygons = mkArchive {
+    url = "https://osmdata.openstreetmap.de/download/water-polygons-split-3857.zip";
+    archive_time = "20241201150707";
+    hash = "sha256-YV3KCcVGZoSrZVjASUCPEcTCn0C39l0pG4gBl5AJm5M=";
+  };
+  icesheet_polygons = mkArchive {
+    url = "https://osmdata.openstreetmap.de/download/antarctica-icesheet-polygons-3857.zip";
+    archive_time = "20241201150705";
+    hash = "sha256-QvaegTQuiju8B8IyT6qq3QDjBWk2D02bu6g07G8VPMA=";
+  };
+  icesheet_outlines = mkArchive {
+    url = "https://osmdata.openstreetmap.de/download/antarctica-icesheet-outlines-3857.zip";
+    archive_time = "20241201150707";
+    hash = "sha256-d09olTATtgLO4VOAAnbO6XOrSm89DOLbSzP4id7njtU=";
+  };
+  ne_110m_admin_0_boundary_lines_land = mkArchive {
+    url = "https://naturalearth.s3.amazonaws.com/110m_cultural/ne_110m_admin_0_boundary_lines_land.zip";
+    archive_time = "20241201150708";
+    hash = "sha256-uylUmBVwSYpLQ0T46D+lTKpYWOFegACUESQH0r0vlA0=";
+  };
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "openstreetmap-carto";
+  version = "5.9.0-unstable-2024-11-27";
+
+  src = fetchFromGitHub {
+    owner = "gravitystorm";
+    repo = "openstreetmap-carto";
+    rev = "5791e79934164f8d5ff73460ad993f3d9f64222c";
+    hash = "sha256-0kPgX09oJprnTDnXdLhse8VWz/crjN5d40yKS0WhsHc=";
+  };
+
+  postPatch = ''
+    substituteInPlace external-data.yml \
+      --replace-fail           '${simplified_water_polygons.url}' 'file://${simplified_water_polygons.archive}' \
+      --replace-fail                      '${water_polygons.url}' 'file://${water_polygons.archive}' \
+      --replace-fail                   '${icesheet_polygons.url}' 'file://${icesheet_polygons.archive}' \
+      --replace-fail                   '${icesheet_outlines.url}' 'file://${icesheet_outlines.archive}' \
+      --replace-fail '${ne_110m_admin_0_boundary_lines_land.url}' 'file://${ne_110m_admin_0_boundary_lines_land.archive}'
+
+    if grep http external-data.yml; then
+      echo 'Not all URLs were patched!'
+      exit 1
+    fi
+
+    substituteInPlace scripts/get-external-data.py \
+      --replace-fail \
+        'ogrcommand = ["ogr2ogr",' \
+        'ogrcommand = ["${lib.getExe' gdal "ogr2ogr"}",' \
+      --replace-fail \
+        '#!/usr/bin/env python3' \
+        '#!${lib.getExe get-external-data_env}'
+
+    substituteInPlace scripts/generate_road_colours.py scripts/generate_shields.py scripts/generate_unpaved_patterns.py scripts/indexes.py \
+      --replace-fail \
+        '#!/usr/bin/env python3' \
+        '#!${lib.getExe generate_env}'
+  '';
+
+  outputs = [
+    "out"
+    "sql"
+    "lua"
+    "get_external_data"
```

I suggest to rename this output.

```suggestion
    "external_data"
```

---

The script is called `get-external-data.py`, so I named the output `get_external_data`. The output does not directly contain the external data but only a script that knows how to "get" it. Do you still think that I should rename it?

---

I am still in favor calling it `external_data`.